Kernel/FileSystem/FileDescription.cpp

```diff
@@ -98,11 +98,7 @@ off_t FileDescription::seek(off_t offset, int whence)
     if (!m_file->is_seekable())
         return -EINVAL;
 
-    auto metadata = this->metadata();
-    if (!metadata.is_valid())
-        return -EIO;
-
-    if (metadata.is_socket() || metadata.is_fifo())
+    if (m_file->is_socket() || m_file->is_fifo())
         return -ESPIPE;
```

Comment:
Yes. There's no valid metadata when you mount the first harddrive in the stage2 of kernel initialization, because we directly sent a `BlockDevice` that isn't tied to any filesystem.

Reply:
Ok great.